src/EventListener/ResponseExceptionListener.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/**
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2015 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\CoreBundle\Exception\ResponseException;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+
+/**
+ * Creates a response from an exception.
+ *
+ * @author Leo Feyer <https://github.com/leofeyer>
+ */
+class ResponseExceptionListener
+{
+    /**
+     * Sets the response from the exception.
+     *
+     * @param GetResponseForExceptionEvent $event The event object
+     */
+    public function onKernelException(GetResponseForExceptionEvent $event)
+    {
+        $exception = $event->getException();
+
+        if (!$exception instanceof ResponseException) {
+            return;
+        }
+
+        $response = $exception->getResponse();
+
+        if (!$response instanceof Response) {
+            return;
+        }
+
+        $event->setResponse($response);
```

Comment:
Has been restored in 6e9d445e837f429efc6d8a2cabe6705689c66ddc and improved in 01812bf87610ddcddf1acc0a9d60103082c806dd